### PR TITLE
EVG-2500 EVG-2568 Add project-ref setting to flag projects that are using the webhook/add to project admin page

### PR DESCRIPTION
**Pull request by richardsamuels**

### CODE REVIEW

**Comment by richardsamuels** [comment]
Screenshot:
a

File: service/project.go [hunk 1]
```diff
@@ -236,13 +232,21 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
...
 	projectVars.Vars = responseRef.ProjVarsMap
 	projectVars.PrivateVars = responseRef.PrivateVars
 	projectVars.PatchDefinitions = responseRef.PatchDefinitions
+
+	err = projectRef.Upsert()
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, err)
+		return
+	}
+
```
Comment (tychoish) [inline service/project.go]: nit: probably makes more sense to move this bloc before line 239

File: units/repotracker.go [hunk 2]
```diff
@@ -90,6 +90,16 @@ func (j *repotrackerJob) Run() {
 		return
 	}
 
+	if !ref.TracksPushEvents {
```
Comment (tychoish) [inline units/repotracker.go]: isn't it a programmer error if we run this job for this project?

Comment (richardsamuels) [reply units/repotracker.go]: As written, all push events create amboy jobs, and the job is responsible for determining which pushevents should trigger repotracker by looking up the ref. Would you prefer I move the ref validation to the PushEvent handler?

File: service/project.go (hunk 1 above)
Comment (richardsamuels) [reply service/project.go]: done

File: units/repotracker.go (hunk 2 above)
Comment (tychoish) [reply units/repotracker.go]: I think in that case, it's not really an error? just log a debug and don't even add errors?

**Pull request merged**